### PR TITLE
TT-1758 passing command line arguments for test tool

### PR DESCRIPTION
**Pull request by SKeerthana**

### CODE REVIEW

File: verify-matching-service-test-tool/src/main/java/uk/gov/ida/verifymatchingservicetesttool/configurations/ConfigurationReader.java [hunk 1]
```diff
@@ -9,22 +9,26 @@
...
+            return new ObjectMapper(new YAMLFactory()).readValue(new File(getConfigurationFolderLocation(configFileLocation)),
+                    ApplicationConfiguration.class);
         } catch (IOException e) {
             throw new RuntimeException(e);
         }
     }
 
-    private static String getConfigurationFolderLocation() {
+    private static String getConfigurationFolderLocation(String configFileLocation) {
+        return configFileLocation != null ? configFileLocation : getDefaultConfigLocation();
```
Comment (andy-paine) [inline verify-matching-service-test-tool/src/main/java/uk/gov/ida/verifymatchingservicetesttool/configurations/ConfigurationReader.java]: I think `Optional.ofNullable(configFileLocation).orElseGet(() => getDefaultConfigLocation())` is more idiomatic Java 8

File: verify-matching-service-test-tool/src/main/java/uk/gov/ida/verifymatchingservicetesttool/utils/CommandLineOptionParser.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    private static String getAbsoluteFilePath(String fileName) {
+        return fileName != null ? System.getProperty("user.dir") + File.separator + fileName : fileName;
+    }
+
+    private static boolean checkIfFileDoesNotExists(String fileName) {
+        return !Files.exists(Paths.get(fileName));
+    }
+
+    private static boolean checkIfFileDoesNotExists(String fileName, FolderName folderName) {
+        return !Files.exists(Paths.get(fileName + File.separator + folderName.getValue()));
```
Comment (andy-paine) [inline verify-matching-service-test-tool/src/main/java/uk/gov/ida/verifymatchingservicetesttool/utils/CommandLineOptionParser.java]: I think there is an overload in `Paths` that lets you avoid using File.separator (you just give it each file path element and it concatenates them for you)

File: verify-matching-service-test-tool/src/main/java/uk/gov/ida/verifymatchingservicetesttool/utils/CommandLineOptionParser.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+        return new CommandLineOptionValue(exampleScenariosFolderLocation, configFileLocation);
+    }
+
+    private static String getAbsoluteFilePath(String fileName) {
+        return fileName != null ? System.getProperty("user.dir") + File.separator + fileName : fileName;
+    }
+
+    private static boolean checkIfFileDoesNotExists(String fileName) {
+        return !Files.exists(Paths.get(fileName));
```
Comment (andy-paine) [inline verify-matching-service-test-tool/src/main/java/uk/gov/ida/verifymatchingservicetesttool/utils/CommandLineOptionParser.java]: `DoesNotExists` reads a bit weirdly 🤔

File: verify-matching-service-test-tool/src/main/java/uk/gov/ida/verifymatchingservicetesttool/configurations/ConfigurationReader.java (hunk 1 above)
Comment (SKeerthana) [reply verify-matching-service-test-tool/src/main/java/uk/gov/ida/verifymatchingservicetesttool/configurations/ConfigurationReader.java]: Done

File: verify-matching-service-test-tool/src/main/java/uk/gov/ida/verifymatchingservicetesttool/utils/CommandLineOptionParser.java (hunk 2 above)
Comment (SKeerthana) [reply verify-matching-service-test-tool/src/main/java/uk/gov/ida/verifymatchingservicetesttool/utils/CommandLineOptionParser.java]: Done

File: verify-matching-service-test-tool/src/main/java/uk/gov/ida/verifymatchingservicetesttool/utils/CommandLineOptionParser.java (hunk 3 above)
Comment (SKeerthana) [reply verify-matching-service-test-tool/src/main/java/uk/gov/ida/verifymatchingservicetesttool/utils/CommandLineOptionParser.java]: Done

**Pull request merged**